src/databricks/labs/blueprint/paths.py

```diff
@@ -448,17 +441,296 @@ def match(self, path_pattern, *, case_sensitive=None):
+            msg = f"Non-relative patterns are unsupported: {pattern!s}"
+            raise NotImplementedError(msg)
+        pattern_parts = parsed_pattern._path_parts  # pylint: disable=protected-access
+        if ".." in pattern_parts:
+            msg = f"Parent traversal is not supported: {pattern!s}"
+            raise ValueError(msg)
+        if os.fspath(pattern)[-1] == self.parser.sep:
+            pattern_parts = (*pattern_parts, "")
+        return pattern_parts
+
+    def glob(
+        self: P, pattern: str | bytes | os.PathLike, *, case_sensitive: bool | None = None
+    ) -> Generator[P, None, None]:
+        pattern_parts = self._prepare_pattern(pattern)
+        if case_sensitive is None:
+            case_sensitive = True
+        selector = _Selector.parse(pattern_parts, case_sensitive=case_sensitive)
+        yield from selector(self)
+
+    def rglob(
+        self: P, pattern: str | bytes | os.PathLike, *, case_sensitive: bool | None = None
+    ) -> Generator[P, None, None]:
+        pattern_parts = ("**", *self._prepare_pattern(pattern))
+        if case_sensitive is None:
+            case_sensitive = True
+        selector = _Selector.parse(pattern_parts, case_sensitive=case_sensitive)
+        yield from selector(self)
+
+    @abstractmethod
+    def as_uri(self) -> str: ...
+
+    @abstractmethod
+    def as_fuse(self) -> Path: ...
+
+    @abstractmethod
+    def exists(self, *, follow_symlinks: bool = True) -> bool: ...
+
+    @abstractmethod
+    def mkdir(self, mode: int = 0o600, parents: bool = True, exist_ok: bool = True) -> None: ...
+
+    @abstractmethod
+    def rmdir(self, recursive: bool = False) -> None: ...
+
+    @abstractmethod
+    def unlink(self, missing_ok: bool = False) -> None: ...
+
+    @abstractmethod
+    def open(
+        self,
+        mode: str = "r",
+        buffering: int = -1,
+        encoding: str | None = None,
+        errors: str | None = None,
+        newline: str | None = None,
+    ): ...
+
+    @abstractmethod
+    def is_dir(self) -> bool: ...
+
+    @abstractmethod
+    def is_file(self) -> bool: ...
+
+    @abstractmethod
+    def expanduser(self: P) -> P: ...
+
+    @abstractmethod
+    def iterdir(self: P) -> Generator[P, None, None]: ...
+
+
+class DBFSPath(_DatabricksPath):
+    """Experimental implementation of pathlib.Path for DBFS paths."""
+
+    __slots__ = (
+        # The cached _file_info value for the instance.
+        "_cached_file_info",
+    )
+    _cached_file_info: FileInfo
+
+    @classmethod
+    def _from_file_info(cls, ws: WorkspaceClient, file_info: FileInfo) -> DBFSPath:
+        """Special (internal-only) constructor that creates an instance based on FileInfo."""
+        if not file_info.path:
+            msg = f"Cannot initialise without file path: {file_info}"
+            raise ValueError(msg)
+        path = cls(ws, file_info.path)
+        path._cached_file_info = file_info
+        return path
+
+    as_uri = _na("as_uri")
```

Comment:
okay, as_uri can stay abstract

Reply:
It can't stay abstract: instantiating without implementing `@abstractemethod` triggers an error.

I've moved this `_na()` to the base class and dropped the `@abstractmethod` declaration. This means now that it defaults to not-supported, with `WorkspacePath` explicitly implementing it.